apps/spruce/src/pages/image/PackagesTable/index.tsx

```diff
@@ -0,0 +1,135 @@
+import { useMemo, useRef, useState, useEffect } from "react";
+import { useQuery } from "@apollo/client";
+import {
+  useLeafyGreenTable,
+  LGColumnDef,
+  ColumnFiltersState,
+  PaginationState,
+} from "@leafygreen-ui/table";
+import { BaseTable } from "components/Table/BaseTable";
+import { DEFAULT_PAGE_SIZE } from "constants/index";
+import { useToastContext } from "context/toast";
+import {
+  Package,
+  ImagePackagesQuery,
+  ImagePackagesQueryVariables,
+} from "gql/generated/types";
+import { IMAGE_PACKAGES } from "gql/queries";
+
+type PackagesTableProps = {
+  imageId: string;
+};
+
+export const PackagesTable: React.FC<PackagesTableProps> = ({ imageId }) => {
+  const dispatchToast = useToastContext();
+  const [pagination, setPagination] = useState<PaginationState>({
+    pageIndex: 0,
+    pageSize: DEFAULT_PAGE_SIZE,
+  });
+  const [columnFilters, setColumnFilters] = useState<ColumnFiltersState>([]);
+  const { data: packagesData, loading } = useQuery<
+    ImagePackagesQuery,
+    ImagePackagesQueryVariables
+  >(IMAGE_PACKAGES, {
+    variables: {
+      imageId,
+      opts: {
+        page: pagination.pageIndex,
+        limit: pagination.pageSize,
+        name: columnFilters.find((filter) => filter.id === "name")
+          ?.value as string,
+      },
+    },
+    onError(err) {
+      dispatchToast.error(
+        `There was an error loading image packages: ${err.message}`,
+      );
+    },
+  });
+
+  const packageEntries = useMemo(
+    () => packagesData?.image?.packages.data ?? [],
+    [packagesData?.image?.packages.data],
+  );
+
+  const numPackages = useMemo(
+    () =>
+      packagesData?.image?.packages.filteredCount ??
+      packagesData?.image?.packages.totalCount,
+    [
+      packagesData?.image?.packages.filteredCount,
+      packagesData?.image?.packages.totalCount,
+    ],
+  );
+
+  const tableContainerRef = useRef<HTMLDivElement>(null);
+  const table = useLeafyGreenTable<Package>({
+    columns,
+    data: packageEntries,
+    containerRef: tableContainerRef,
+    defaultColumn: {
+      enableColumnFilter: false,
+    },
+    manualPagination: true,
+    manualFiltering: true,
+    rowCount: numPackages,
+    state: {
+      pagination,
+      columnFilters,
+    },
+    onColumnFiltersChange: setColumnFilters,
+    onPaginationChange: setPagination,
+  });
+
+  // Auto-redirect user to first page when filter applied.
+  useEffect(() => {
+    if (
+      table &&
+      packagesData?.image?.packages.filteredCount &&
+      packagesData?.image?.packages.totalCount &&
+      packagesData?.image?.packages.filteredCount <
+        packagesData?.image?.packages.totalCount
+    ) {
+      table.firstPage();
+    }
+  }, [
+    table,
+    packagesData?.image?.packages.filteredCount,
+    packagesData?.image?.packages.totalCount,
+  ]);
```

Comment:
I think that the logic to reset the page should be inside of `BaseTable`, not outside of it. If the logic was kept outside, that means every time we use this component we would need to rewrite this code every time.

In `BaseTable`, the filter logic is present in the headers. Can we reset the page when a filter is applied, somewhere in the code in `BaseTable`?